tools/report-converter/codechecker_report_converter/analyzers/sanitizers/address/parser.py

```diff
@@ -25,3 +28,78 @@ class Parser(SANParser):
+        "bad-__sanitizer_get_allocated_size": re.compile(
+            "attempting to call __sanitizer_get_allocated_size() for pointer "
+            "which is not owned: .+"),
+        "calloc-overflow": re.compile(
+            "calloc parameters overflow: count * size (.+ * .+) cannot be "
+            "represented in type size_t (thread .+)"),
+        "reallocarray-overflow": re.compile(
+            "reallocarray parameters overflow: count * size (.+ * .+) cannot "
+            "be represented in type size_t (thread .+)"),
+        "pvalloc-overflow": re.compile(
+            "pvalloc parameters overflow: size 0x.+ rounded up to system page "
+            "size 0x.+ cannot be represented in type size_t (thread .+)"),
+        "invalid-allocation-alignment": re.compile(
+            "invalid allocation alignment: .+, alignment must be a power of "
+            "two (thread .+)"),
+        "invalid-aligned-alloc-alignment": re.compile(
+            "invalid alignment requested in aligned_alloc: .+ the requested "
+            "size 0x.+ must be a multiple of alignment (thread .+)"),
+        "invalid-posix-memalign-alignment": re.compile(
+            "invalid alignment requested in posix_memalign: .+, alignment "
+            "must be a power of two and a multiple of sizeof(void*) == .+ "
+            "(thread .+)"),
+        "allocation-size-too-big": re.compile(
+            "requested allocation size 0x.+ (0x.+ after adjustments for "
+            "alignment, red zones etc.) exceeds maximum supported size of "
+            "0x.+ (thread .+)"),
+        "rss-limit-exceeded": re.compile(
+            "specified RSS limit exceeded, currently set to "
+            "soft_rss_limit_mb=.+"),
+        "out-of-memory": re.compile(
+            "allocator is trying to allocate 0x.+ bytes"),
+        "string-function-memory-ranges-overlap": re.compile(
+            ".+: memory ranges .+ and .+ overlap"),
+        "negative-size-param": re.compile(
+            "negative-size-param: (size=.+)"),
+        "bad-__sanitizer_annotate_contiguous_container": re.compile(
+            "bad parameters to __sanitizer_annotate_contiguous_container:"),
+        "bad-__sanitizer_annotate_double_ended_contiguous_container":
+        re.compile(
+            "bad parameters to "
+            "__sanitizer_annotate_double_ended_contiguous_container:"),
+        "odr-violation": re.compile("odr-violation (.+):"),
+        "invalid-pointer-pair": re.compile("invalid-pointer-pair: .+ .+")
+    }
+
+    def parse_sanitizer_message(
+        self,
+        it: Iterator[str],
+        line: str
+    ) -> Tuple[Optional[Report], str]:
+        report, line = super().parse_sanitizer_message(it, line)
+
+        if not report:
+            return report, line
+
+        for check, pattern in self.checks.items():
+            if pattern.search(report.message):
+                report.checker_name = f"{self.checker_name}.{check}"
+                break
+
```

Comment:
Do we not expect sanitized messages to change every once in a while? Especially to expand it with more information? Since this is on the client side, if this heuristic fails due to a change in the message, we won't recognize that those two reports are in fact the same. Or, at least, that is my theory... checker names do contribute to the bug hash, right?

Reply:
To be frank, I'm not opposing this patch on this ground, I'm just pointing out the risks, though I'm sure you are perfectly aware of them. Is there no other way to make this any smarter?

Reply:
That's right, this implementation relies on the checker message which may change eventually. I don't know how frequent even this is. I can't come up with better heuristics, though. In the worst case the message pattern will not match and it falls back to the sanitizer's name in which case we can adapt the changes in the heuristics.